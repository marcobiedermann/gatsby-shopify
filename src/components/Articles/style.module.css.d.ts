declare const styles: {
  readonly articles: string;
};
export = styles;
