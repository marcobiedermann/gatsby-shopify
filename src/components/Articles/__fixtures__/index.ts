import { ArticlesProps } from '..';
import article from '../../Article/__fixtures__';

const articles: ArticlesProps = {
  articles: [article],
};

export default articles;
