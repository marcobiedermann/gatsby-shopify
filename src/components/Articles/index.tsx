import React, { FC } from "react"
import Article, { ArticleProps } from "../Article"

export interface ArticlesProps {
  articles: ArticleProps[]
}

const Articles: FC<ArticlesProps> = props => {
  const { articles } = props

  return (
    <ol>
      {articles.map(article => (
        <li key={article.shopifyId}>
          <Article {...article} />
        </li>
      ))}
    </ol>
  )
}

export default Articles
