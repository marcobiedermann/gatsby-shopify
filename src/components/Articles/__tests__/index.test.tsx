import { render } from '@testing-library/react';
import React from 'react';
import Articles from '..';
import articles from '../__fixtures__';

describe('articles component', () => {
  it('should render correctly', () => {
    const { container } = render(<Articles {...articles} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
