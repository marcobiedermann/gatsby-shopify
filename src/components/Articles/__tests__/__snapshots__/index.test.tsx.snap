// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articles component should render correctly 1`] = `
<ol>
  <li>
    <article>
      <h3>
        <a
          href="/blogs/news/article-title"
        >
          Article Title
        </a>
      </h3>
      <div />
      <p>
        <a
          href="/blogs/news/article-title"
        >
          Read more
        </a>
      </p>
    </article>
  </li>
</ol>
`;
